root/static/scripts/edit/URLCleanup.js

```diff
@@ -1317,6 +1317,21 @@ const CLEANUPS = {
       return {result: false};
     },
   },
+  'dnb': {
+    match: [
+      new RegExp('^(https?://)?([^/]+\\.)?d-nb\\.info', 'i'),
+      new RegExp('^(https?://)?([^/]+\\.)?dnb\\.de', 'i'),
+    ],
+    type: LINK_TYPES.otherdatabases,
+    clean: function (url) {
+      url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?dnb\.de\/opac\.htm\?.*\bquery=nid%3D(\d+).*$/, 'http://d-nb.info/gnd/$1');
+      url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?dnb\.de\/opac\.htm\?.*\bquery=idn%3D(\d+).*$/, 'http://d-nb.info/$1');
+      return url;
+    },
+    validate: function (url, id) {
+      return {result: id === LINK_TYPES.otherdatabases.artist};
```

Comment:
You added the `idn` matching for releases, but this still only allows these links on artists :) You should have two checks:
If it matches `http://d-nb.info/gnd/(id)` then it's allowed for artist, otherwise not.
If it matches `http://d-nb.info/(id)` then it's allowed for release, otherwise not.